File: app/schemas/manual_assets.py
from typing import Optional
from pydantic import BaseModel


class FdAsset(BaseModel):
    id: int
    label: Optional[str] = None
    principal: float
    interest_rate: float
    start_date: Optional[str] = None
    maturity_date: Optional[str] = None
    current_value: float
    maturity_value: float
    is_emergency_fund: bool


class SimpleAsset(BaseModel):
    id: int
    label: Optional[str] = None
    current_value: float


class ForeignEquityAsset(BaseModel):
    id: int
    label: str
    value_usd: float
    invested_usd: float
    value_inr: float


class ManualAssetsSummary(BaseModel):
    fds: list[FdAsset]
    ppf: Optional[SimpleAsset] = None
    nps: Optional[SimpleAsset] = None
    cash: Optional[SimpleAsset] = None
    foreign_equities: list[ForeignEquityAsset] = []
    total_fd: float
    emergency_total: float
    total_ppf: float
    total_nps: float
    total_cash: float
    total_foreign_equity_usd: float = 0.0
    total_foreign_equity_inr: float = 0.0
    usdinr_rate: float = 85.0
    total_manual: float

File: frontend/src/types/manualAssets.ts
export interface FdAsset {
  id: number
  label: string | null
  principal: number
  interest_rate: number
  start_date: string | null
  maturity_date: string | null
  current_value: number
  maturity_value: number
  is_emergency_fund: boolean
}

export interface SimpleAsset {
  id: number
  label: string | null
  current_value: number
}

export interface ForeignEquityAsset {
  id: number
  label: string
  value_usd: number
  invested_usd: number
  value_inr: number
}

export interface ManualAssetsSummary {
  fds: FdAsset[]
  ppf: SimpleAsset | null
  nps: SimpleAsset | null
  cash: SimpleAsset | null
  foreign_equities: ForeignEquityAsset[]
  total_fd: number
  emergency_total: number
  total_ppf: number
  total_nps: number
  total_cash: number
  total_foreign_equity_usd: number
  total_foreign_equity_inr: number
  usdinr_rate: number
  total_manual: number
}

export interface UsdinrInfo {
  rate: number
  source: string | null
  fetched_at: string | null
}
